packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,287 @@
+<template>
+  <KPop
+    class="analytics-timepicker"
+    placement="bottomStart"
+    width="auto"
+    hide-caret
+  >
+    <KButton
+      :is-rounded="false"
+      size="large"
+      class="timepicker-input"
+    >
+      {{ selectedTimeframe.timeframeText }}
+    </KButton>
+    <template #content>
+      <KSegmentedControl
+        v-model="mode"
+        :options="[
+          { label: 'Custom', value: 'custom' },
+          { label: 'Relative', value: 'relative' }
+        ]"
+        class="w-100 mb-5"
+        data-testid="analytics-time-toggle"
+      />
+      <DatePicker
+        v-if="showCalendar"
+        :model-config="modelConfig"
+        v-model="selectimedTimeRange"
+        mode="dateTime"
+        is-range
+        is-expanded
+      />
+      <div
+        v-else
+        class="relative-frames d-flex justify-content-start"
+      >
+        <KButton
+          v-for="(item, index) in allowedTimePeriods"
+          :key="`time-${item.timeframeLength}-${index}`"
+          :is-rounded="false"
+          :class="{'selected-option': item.timeframeText === selectedTimeframe.timeframeText}"
+          :data-testid="'select-timeframe-' + item.timeframeLength"
+          appearance="outline"
+          size="medium"
+          @click="changeTimeframe(item)"
+        >
+          {{ item.timeframeText }}
+        </KButton>
+      </div>
+    </template>
+    <template #footer>
+      <div class="d-flex justify-content-end">
+        <KButton
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="clearSelection()"
+        >
+          Clear
+        </KButton>
+        <KButton
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="submitTimeFrame()"
+        >
+          Apply
+        </KButton>
+      </div>
+    </template>
+  </KPop>
+</template>
+
+<script>
+// import { DatePicker } from 'v-calendar'
+import KButton from '@kongponents/kbutton/KButton.vue'
+import KPop from '@kongponents/kpop/KPop.vue'
+import KSegmentedControl from '@kongponents/ksegmentedcontrol/KSegmentedControl.vue'
+import { roundToNearestMinutes, getUnixTime } from 'date-fns'
+
+const allowedTimePeriods = [
```

Comment:
Can we move this `allowedTimePeriods` into a `.ts` file within the `KDateTimePciker/` directory and just import it here? Same if you end up with any other config objects, etc.

Reply:
I totally missed this suggestion.  This solves my need to import `date-fns` directly into the markdown file. Stay tuned!